src/test/java/com/google/api/gax/grpc/UnaryCallableTest.java

```diff
@@ -458,25 +460,72 @@ public Integer extractPageSize(Integer payload) {
     }
   }
 
-  private class PagedListResponse extends PagedListResponseImpl<Integer, List<Integer>, Integer> {
-    public PagedListResponse(
-        UnaryCallable<Integer, List<Integer>> callable,
-        PagedListDescriptor<Integer, List<Integer>, Integer> pageDescriptor,
-        Integer request,
-        CallContext context) {
-      super(callable, pageDescriptor, request, context);
+  private static class ListIntegersPagedResponse
+      extends AbstractPagedListResponse<
+          Integer, List<Integer>, Integer, ListIntegersPage, ListIntegersSizedPage> {
+
+    protected ListIntegersPagedResponse(ListIntegersPage page) {
+      super(page, ListIntegersSizedPage.createEmptyCollection());
+    }
+
+    public static ListIntegersPagedResponse create(
+        PageContext<Integer, List<Integer>, Integer> context, List<Integer> response) {
+      ListIntegersPage page = new ListIntegersPage(context, response);
```

Comment:
It seems like it would be better to lift the page construction up to `PagedFactory.callAndCreateResponse` - the constructed page can then just be passed into here.